### PR TITLE
buildbot: 0.9.0.post1 -> 0.9.3

### PR DESCRIPTION
**Pull request by nand0p**
- Fixes unneeded patching
- Adds worker to build inputs now needed for tests
- Replaces enableworker option with worker configuration module
- Openssh required for tests
- Fixes worker hardcoded paths
- Tested on Nixos Unstable

### CODE REVIEW

**Comment by nand0p** [comment]
@FRidh @copumpkin ready_for_review

File: pkgs/development/tools/build-managers/buildbot/plugins.nix
Comment (FRidh) [inline pkgs/development/tools/build-managers/buildbot/plugins.nix]: you might want to check out the new `pythonPackages.fetchPypi` function.

Comment (nand0p) [inline pkgs/development/tools/build-managers/buildbot/plugins.nix]: seems to get a 404 with fetchPypi:



`-    src = fetchurl {`
`-      url = "https://pypi.python.org/packages/02/d0/fc56ee27a09498638a47dcc5637ee5412ab7a67bfb4b3ff47e041f3d7b66/${name}-py2-none-any.whl";`
`-      sha256 = "14ghch67k6090736n89l401swz7r9hnk2zlmdb59niq8lg7dyg9q";`
`+    src = pythonPackages.fetchPypi {`
`+      inherit pname version format;`
`+      sha256 = "1yggg6mcykcnv41srl2sp2zwx2r38vb6a8jgxh1a4825mspm2jf7";
`
this results in:

> curl: (22) The requested URL returned error: 404 
> error: cannot download buildbot_www-0.9.3-py2.py3-none-any.whl from any mirror
> builder for ‘/nix/store/3kscl522lv3n73dlcskjlr4rxckaldkb-buildbot_www-0.9.3-py2.py3-none-any.whl.drv’ failed with exit code 1

Comment (FRidh) [inline pkgs/development/tools/build-managers/buildbot/plugins.nix]: That's because the package is named `buildbot-www` and not `buildbot_www`.
This is a bit frustrating, Nixpkgs guidelines say the name should be `buildbot_www` but in pythonPackages I push for the original name.

Comment (nand0p) [inline pkgs/development/tools/build-managers/buildbot/plugins.nix]: looks like same error with `buildbot-www`

> curl: (22) The requested URL returned error: 404 
> error: cannot download buildbot-www-0.9.3-py2.py3-none-any.whl from any mirror
> builder for ‘/nix/store/k9pxvisj91gkydfhkhypw4czyzvxfhw4-buildbot-www-0.9.3-py2.py3-none-any.whl.drv’ failed with exit code 1

Comment (FRidh) [inline pkgs/development/tools/build-managers/buildbot/plugins.nix]: Oh, interesting, the name is `buildbot_www-0.9.3-py2-none-any.whl `, so
```
fetchPypi {
  pname = "buildbot_www"; # or maybe inherit
  inherit version;
  python = "py2"; # Apparently its only compatible with Python 2.
  sha256 =  "c74951afae4520a202ec4f2265d646238bcebfb85ad0ac03d9964dcfaa79ef79";
};
```
should work.

In case any of the optional arguments has to be changed its perhaps not worth using this function instead of just using the link. What you choose is up to you.

**Comment by nand0p** [comment]
@FRidh updated `buildbot_www` with `fetchPypi`. this is working now.

File: pkgs/development/tools/build-managers/buildbot/worker.nix
Comment (FRidh) [inline pkgs/development/tools/build-managers/buildbot/worker.nix]: why is this needed? Is the current wrapper not working?

Comment (nand0p) [inline pkgs/development/tools/build-managers/buildbot/worker.nix]: that appears to be so.  if you look at the changes in the bb master expression, you can see the wrapper issue there as well:

`-    mv -v $out/bin/buildbot $out/bin/.wrapped-buildbot`
`-    echo "#!/bin/sh" > $out/bin/buildbot`
`-    echo "export PYTHONPATH=$PYTHONPATH" >> $out/bin/buildbot`
`-    echo "exec $out/bin/.wrapped-buildbot \"\$@\"" >> $out/bin/buildbot`
`-    chmod -c 555 $out/bin/buildbot`
`+    makeWrapper $out/bin/.buildbot-wrapped $out/bin/buildbot --set PYTHONPATH "$PYTHONPATH"`

it looks like without this wrapper in worker, worker was working on centos, but not nixos.

File: pkgs/development/python-modules/incremental/default.nix
Comment (FRidh) [inline pkgs/development/python-modules/incremental/default.nix]: Don't pass in `pkgs`, nor `pythonpackages`. Instead, have the packages as parameters.

File: pkgs/development/python-modules/treq/default.nix
Comment (FRidh) [inline pkgs/development/python-modules/treq/default.nix]: same here

**Comment by nand0p** [comment]
@FRidh fixed up passing packages as parameters. tested and working.

**Comment by FRidh** [comment]
@nand0p 
In case a Python package is called from `python-packages.nix`, the package expression should have as parameters the individual packages that it needs.

In case a Python application is called from `all-packages.nix`, the package expression should have as as parameters the Python interpreter (`python`, so you use `python.pkgs` or `python.withPackages`), or the Python package set `pythonXXPackages`.

**Comment by nand0p** [comment]
@FRidh fixed

**Review by FRidh — APPROVED** [review]

**Comment by nand0p** [comment]
@FRidh @copumpkin looks like CI is failing unrelated.  can we override and merge?


`$ ./maintainers/scripts/travis-nox-review-pr.sh nixpkgs-verify nixpkgs-manual nixpkgs-tarball nixpkgs-unstable`
`=== Verifying that nixpkgs evaluates...`
`error: value is a string while a list was expected, at /home/travis/build/NixOS/nixpkgs/pkgs/development/interpreters/python/mk-python-derivation.nix:93:19`
`The command "./maintainers/scripts/travis-nox-review-pr.sh nixpkgs-verify nixpkgs-manual nixpkgs-tarball nixpkgs-unstable" exited with 1.`

File: pkgs/development/python-modules/treq/default.nix
Comment (FRidh) [inline pkgs/development/python-modules/treq/default.nix]: tests aren't run

File: pkgs/development/python-modules/treq/default.nix
Comment (copumpkin) [inline pkgs/development/python-modules/treq/default.nix]: Isn't this expected to be a list?

Comment (FRidh) [inline pkgs/development/python-modules/treq/default.nix]: yep

**Comment by nand0p** [comment]
@FRidh updated `treq` to explicitly disable tests and commented with the error.  I have another PR in the works which addresses broken tests with buildbot dependencies (including treq), so hopefully this will not block merging, and we can fix iteratively.

**Comment by copumpkin** [comment]
Looks good to me. @FRidh any objections to merging?

**Comment by copumpkin** [comment]
@FRidh oh, I missed that you'd already approved this earlier. I'll just merge now, since the only remaining issues seem to be fixed.

**Pull request merged**
